src/java/org/apache/cassandra/db/marshal/BytesType.java

```diff
@@ -91,6 +91,40 @@ public boolean isValueCompatibleWithInternal(AbstractType<?> otherType)
         return true;
     }
 
+    @Override
+    public boolean isSerializationCompatibleWith(AbstractType<?> previous)
+    {
+        // BytesType can read anything, but for serialization compatibility (i.e., when dropping
+        // and re-adding a column), we need to be very restrictive. We cannot be serialization
+        // compatible with geometric types or date ranges as they have different wire formats.
+        if (previous instanceof AbstractGeometricType || previous instanceof DateRangeType)
+            return false;
+
+        // BytesType is serialization-compatible with itself and with a specific set of types
+        // that have compatible serialization semantics, even though they may have different
+        // fixed-length properties.
+        if (this == previous)
+            return true;
+
+        if (isMultiCell() != previous.isMultiCell())
+            return false;
+
+        // BytesType is serialization-compatible only with specific types that maintain
+        // compatible serialization semantics (not with all value-compatible types).
+        // Specifically, it's NOT compatible with IEEE 754 floating point types or
+        // fixed-size integer types like Int32Type/LongType.
+        return previous instanceof SimpleDateType
+            || previous instanceof ByteType
+            || previous instanceof AsciiType
+            || previous instanceof TimeType
+            || previous instanceof UTF8Type
+            || previous instanceof InetAddressType
+            || previous instanceof DurationType
+            || previous instanceof ShortType
+            || previous instanceof IntegerType
+            || previous instanceof DecimalType;
+    }
```

Comment:
what's with this addition (and TimeType), which isn't found upstream ?

Reply:
Like https://github.com/apache/cassandra/blob/trunk/test/unit/org/apache/cassandra/cql3/validation/operations/AlterTest.java#L71 shouldn't be allowed to pass, because then you'll read raw bytes from the frozen collection which is nonsensical.

Reply:
Thanks! 

That ticket also serves as the base for what needs to be ported into `main-5.0` 
https://github.com/riptano/cndb/issues/15778